### ISSUE TITLE
Should generic `arrays` be zero-filled on initialization?

### ISSUE THREAD

**Issue by kgryte** [dstructs/array#3]
Should we standardize how `array` and `typed-arrays` are initialized? Currently, `typed-arrays` are zero-filled, while `arrays` are not.

We could zero-fill by default and provide an `option` for turning off zero-filling. 

Thoughts?

**Comment by Planeshifter** [comment]
Arrays can hold also non-numeric data, so I am not in favor of pre-filling them with zeros. In addition, it is already very easy to create a zero-filled array with the `zeros` function. All in all, I believe this functionality should be left to utility functions.

**Comment by kgryte** [comment]
Yeah, the sparse `array` and non-numeric `array` cases make zero-filling undesirable, which is why I was curious about providing an `option` to turn-off zero-filling. 

I suppose my thinking is whether the default behavior should be uniform across `array` types. Granted, an `array` is not a `typed-array`, but a user may simply opt for `arrays`, not for non-numeric storage, but for a dynamic store. As you say, however, that user could just zero-fill the `array` herself.

**Comment by Planeshifter** [comment]
Another reason for not pre-filling is that this behavior would stand in direct conflict with how vanilla JS arrays are instantiated.

**Comment by kgryte** [comment]
Right. But then again, a user must then be aware that `generic` versus `float64` makes a difference.
